[AI-5181] CloudGen Firewall Integration PR

vinodkumar-sacumen opened the pull request:
### What does this PR do?
This is a initial release PR of Barracuda CloudGen Firewall integration including all the required assets.

### Motivation
This is beta release of Agent based integration and is intended for internal testing before going live. We will raise a separate PR with assets and all necessary information once this PR has been merged.

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Add the `qa/skip-qa` label if the PR doesn't need to be tested during QA.
- [ ] If you need to backport this PR to another branch, you can add the `backport/<branch-name>` label to the PR and it will automatically open a backport PR once this one is merged

sarah-witt commented:
Hi @vinodkumar-sacumen to address the CI errors, could you:
1. Rename `cloudgen_firewall/changelog.d/1.added` to `cloudgen_firewall/changelog.d/20344.added` and write "Initial Release" in it?
2. Fix the errors in `cloudgen_firewall/assets/logs/cloudgen_firewall.yaml` mentioned by the `datadog-assets / validate-logs` bot: https://github.com/DataDog/integrations-core/pull/20344/files#diff-09e03697ffea13ad5f290e53c6ffa8cc2022ada3181c17108226b099c8aa88d3R17
3. Fix the log integration tests by following what the `datadog-assets / validate-logs`
 bot says here: https://github.com/DataDog/integrations-core/pull/20344/files#diff-a2cfcf08dc72d40d7ccae28e683ed40802a6af961791bb60df28ee79f50c2817R1

sarah-witt commented on cloudgen_firewall/manifest.json:
```diff
@@ -0,0 +1,68 @@
+      "events": {
+        "creates_events": false
+      },
+      "metrics": {
+        "prefix": "cloudgen_firewall",
+        "check": "",
+        "metadata_path": "metadata.csv"
+      }
```
```suggestion
```
Does this integration only emit logs? If so, this and the `metadata.csv` can be removed.

baturalp-dd commented:
Hi from logs! There are some linter related error messages. Can you take a look and fix them? Let us know once it’s ready and we’ll be happy to review. Thanks!

vinodkumar-sacumen commented:
> Hi @vinodkumar-sacumen to address the CI errors, could you:
> 
> 1. Rename `cloudgen_firewall/changelog.d/1.added` to `cloudgen_firewall/changelog.d/20344.added` and write "Initial Release" in it?
> 2. Fix the errors in `cloudgen_firewall/assets/logs/cloudgen_firewall.yaml` mentioned by the `datadog-assets / validate-logs` bot: https://github.com/DataDog/integrations-core/pull/20344/files#diff-09e03697ffea13ad5f290e53c6ffa8cc2022ada3181c17108226b099c8aa88d3R17
> 3. Fix the log integration tests by following what the `datadog-assets / validate-logs`
>    bot says here: https://github.com/DataDog/integrations-core/pull/20344/files#diff-a2cfcf08dc72d40d7ccae28e683ed40802a6af961791bb60df28ee79f50c2817R1
> 
> Additionally, could you resolve the merge conflicts in `.github/CODEOWNERS`?

@sarah-witt Hi Sarah, I have fixed No.1 and resolved merge conflicts. For the facet and other pipeline issue, we have already requested for assistance as we are unable to update it (may be due to permission issue).
Requesting you (If you are aware off) to please assist on this. Would be really helpful and get this to closure.

vinodkumar-sacumen commented:
> Hi @vinodkumar-sacumen thanks for using the new template! to address the remaining CI errors, could you:
> 
> 1. Run `ddev validate config cloudgen_firewall -s`
> 2. Fix the errors in `cloudgen_firewall/assets/logs/cloudgen_firewall.yaml` mentioned by the datadog-assets / validate-logs bot: https://github.com/DataDog/integrations-core/pull/20344/files#diff-09e03697ffea13ad5f290e53c6ffa8cc2022ada3181c17108226b099c8aa88d3R17
> 3. Fix the log integration tests by following what the datadog-assets / validate-logs
>    bot says here: https://github.com/DataDog/integrations-core/pull/20344/files#diff-a2cfcf08dc72d40d7ccae28e683ed40802a6af961791bb60df28ee79f50c2817R1

Hi @sarah-witt Thanks for the review comments. We have addressed them. Please have a look and let us know if there is any changes required.

baturalp-dd commented on cloudgen_firewall/assets/logs/cloudgen_firewall.yaml:
```diff
@@ -0,0 +1,511 @@
+      sourceType: attribute
+      target: syslog.timestamp
+      targetType: attribute
+      preserveSource: true
```
We generally do not recommend preserving source field unless there is a backwards compatibility issue. (i.e. updating an existing pipeline, not a new one) Is there a particular reason why it was set to `true` for all attribute-mappers?

vinodkumar-sacumen replied:
Understood. We do not update the existing pipeline. Thanks

baturalp-dd commented on cloudgen_firewall/assets/logs/cloudgen_firewall.yaml:
```diff
@@ -0,0 +1,511 @@
+  filter:
+    query: source:cloudgen_firewall
+  processors:
+    - type: grok-parser
```
I noticed there are 17 different grok-parsers defined at the root level which means they’ll all be evaluated against every log message coming from `source:cloudgen_firewall`. I was wondering if there’s a cleaner way to handle this. Maybe by leveraging distinguishing attributes in the log payloads to route them through specific sub-pipelines? This way we could apply only the relevant grok-parser based on format instead of running all of them each time.

Curious if this has been explored before or if there are any constraints I might be missing.

vinodkumar-sacumen replied:
Basically, logs are of different format and each Grok parser takes only  5 sample inputs for parsing. Hence segregated it based on the log types.

baturalp-dd replied:
Hi, I understand. Is there another field we could use to filter the logs so that different formats can be routed through their own sub-pipelines, each with its own grok parser(s)?

> Hence segregated it based on the log types.

Even though the grok parsers are logically separated by type, they currently sit at the root level. This means every log message is evaluated against all 17 parsers even though typically only one should apply. That’s the main concern with this setup.

If possible, we’d prefer an approach that uses filters to direct logs into sub-pipelines more selectively, avoiding unnecessary parsing overhead. Open to thoughts or other ideas here.

ieguinoa replied:
@vinodkumar-sacumen could you take a look at this please? I'm picking up this review as @baturalp-dd is away but would like to confirm whether there is a way to use a filter instead of having a single layer of many grok parsers.

vinodkumar-sacumen replied:
Hi Baturalp-dd, We tried nested Pipeline. And we see the logs are being parsed in the agent as per expected. But when upadting the PR, the nested pipeline tests are failing for some reason. we have followed and checked all known parameters. Request you to please assist on this. Thanks!

baturalp-dd commented on cloudgen_firewall/assets/logs/cloudgen_firewall_tests.yaml:
```diff
@@ -0,0 +1,170 @@
+id: "cloudgen_firewall"
+tests:
+ -
+  sample: "2025-04-22T12:33:00+05:30 sacumentestlocal sacumentestlocal/box_Auth_access:  Info     sacumentestlocal LOGIN ATTEMPT: boxconfig[8016]: Login localhost_sacumentestlocal from 127.0.0.1 : Allowed."
```
Can you please make sure there is at least 1 sample for each `grok-parser` rule defined above?

buraizu commented on cloudgen_firewall/README.md:
```diff
@@ -0,0 +1,81 @@
+3. Use tcpdump to confirm network traffic. On datadog Agent host:
+    sudo tcpdump -i any port 514
+You should see traffic from the cloudGen Firwewall IP address. if not, check firewall rules between cloudGen and Agent. Confirm the correct protocol (UDP/TCP) is being used on both sides.
+4. Check Live Tail in Datadog. Filter by source and service as defined in conf.yaml.
```
```suggestion
4. Check the Datadog [Live Tail][7] in Datadog for logs from the source and service you defined in the `conf.yaml` file.
```

buraizu commented on cloudgen_firewall/README.md:
```diff
@@ -0,0 +1,81 @@
+    sudo tcpdump -i any port 514
+You should see traffic from the cloudGen Firwewall IP address. if not, check firewall rules between cloudGen and Agent. Confirm the correct protocol (UDP/TCP) is being used on both sides.
+4. Check Live Tail in Datadog. Filter by source and service as defined in conf.yaml.
+5. Create a Test Log on the firewall by triggering an event.
```
```suggestion
5. After following these steps, you can create a test log on the firewall by triggering an event.
```

buraizu commented on cloudgen_firewall/README.md:
```diff
@@ -0,0 +1,81 @@
+You should see traffic from the cloudGen Firwewall IP address. if not, check firewall rules between cloudGen and Agent. Confirm the correct protocol (UDP/TCP) is being used on both sides.
+4. Check Live Tail in Datadog. Filter by source and service as defined in conf.yaml.
+5. Create a Test Log on the firewall by triggering an event.
+6. Check parsing and tagging in datadog.
```
What should a user check for? Are there any [processors](https://docs.datadoghq.com/logs/log_configuration/processors/) that might be helpful for users here?

buraizu commented on cloudgen_firewall/README.md:
```diff
@@ -0,0 +1,81 @@
+4. Check Live Tail in Datadog. Filter by source and service as defined in conf.yaml.
+5. Create a Test Log on the firewall by triggering an event.
+6. Check parsing and tagging in datadog.
+7. Use a Dashboard that is created to track volume over time.
```
```suggestion
```

buraizu replied:
Suggesting to delete this step as it's unrelated to validating the integration. 

With that said, I'm open to including it somewhere else if you think this is important information for users of this integration. If so, would it be possible to provide some guidelines for users who might want to set up such a dashboard, or alternatively link to another page, such as the [log volume control](https://docs.datadoghq.com/observability_pipelines/set_up_pipelines/log_volume_control/) page?

vinodkumar-sacumen replied:
There is no setup required from the user end. Would make sense to delete it.

buraizu commented on cloudgen_firewall/README.md:
````diff
@@ -0,0 +1,81 @@
+          service: <SERVICE_NAME>
+    ```
+
+    Change the `path` and `service` parameter values and configure them for your environment.
````
```suggestion
    Change the `path` and `service` parameter values for your environment.
```

buraizu commented on cloudgen_firewall/README.md:
```diff
@@ -0,0 +1,81 @@
+
+### Events
+
+The cloudgen_firewall integration include recognizable events like messages (eg. login failed, rule hits). These can be surfaced as datadog events with parsing.
```
```suggestion
The cloudgen_firewall integration includes log events such as failed logins and rule hits.
```

buraizu replied:
Since this seems to be a description of the log events generated by the integration, this information would be a good fit for the `### Log collection` section. If the integration does not produce any Datadog events by default, please indicate that `cloudgen_firewall does not include any events`, or, alternatively, provide users with guidance or a link to transform their log events into Datadog events with parsing.

buraizu commented on cloudgen_firewall/README.md:
```diff
@@ -0,0 +1,81 @@
+
+cloudgen_firewall does not include any metrics.
+
+### Log Collection
```
```suggestion
### Log collection
```

shrey4b commented on cloudgen_firewall/images/cloudgen_firewall_overview.png:
The ratio for the images seems to be off, can they be replaced

shrey4b commented on cloudgen_firewall/images/cloudgen_firewall_overview.png:
The ratio for the images seems to be off, can they be replaced

shrey4b commented on cloudgen_firewall/README.md:
Adding a brief summary here of the product as well as what the integration includes (i.e. ready to use dashboards) will be helpful here.

vinodkumar-sacumen replied:
Can we know where?

shrey4b commented on cloudgen_firewall/README.md:
```diff
@@ -0,0 +1,78 @@
+
+### Metrics
+
+cloudgen_firewall does not include any metrics.
```
Can replace with - "The Barracuda CloudGen Firewall integration does not include any metrics."

shrey4b commented on cloudgen_firewall/assets/dashboards/cloudgen_firewall_overview.json:
```diff
@@ -0,0 +1,6908 @@
+                    {
+                        "id": 6163932701500751,
+                        "definition": {
+                            "title": "List of Total loggedIn Users of Auth Access",
```
Should it be List of Total Users Logged In by Auth Access

shrey4b commented on cloudgen_firewall/assets/dashboards/cloudgen_firewall_overview.json:
```diff
@@ -0,0 +1,6908 @@
+                            },
+                            "custom_links": [
+                                {
+                                    "label": "View related Security Signals",
```
View Related Security Signals

shrey4b commented on cloudgen_firewall/assets/dashboards/cloudgen_firewall_overview.json:
```diff
@@ -0,0 +1,6908 @@
+                    {
+                        "id": 1600049059415381,
+                        "definition": {
+                            "title": "All  Events Operative Logs",
```
Double space

shrey4b commented on cloudgen_firewall/assets/dashboards/cloudgen_firewall_overview.json:
```diff
@@ -0,0 +1,6908 @@
+                    {
+                        "id": 5127446461897704,
+                        "definition": {
+                            "title": "Config Changes Event Time Count ",
```
Would it be Config Changes _by_ Event Time Count?

vinodkumar-sacumen replied:
No. It is the time count of config changes events

shrey4b commented on cloudgen_firewall/assets/dashboards/cloudgen_firewall_overview.json:
```diff
@@ -0,0 +1,6908 @@
+                    {
+                        "id": 6629449643271353,
+                        "definition": {
+                            "title": "All  Config Changes Logs ",
```
Double space

shrey4b commented on cloudgen_firewall/assets/dashboards/cloudgen_firewall_overview.json:
```diff
@@ -0,0 +1,6908 @@
+        {
+            "id": 6168621600317357,
+            "definition": {
+                "title": "Config  Daemon Download",
```
Double space

shrey4b commented on cloudgen_firewall/assets/dashboards/cloudgen_firewall_overview.json:
```diff
@@ -0,0 +1,6908 @@
+                    {
+                        "id": 2977175297348942,
+                        "definition": {
+                            "title": "All Config Sync Logs ",
```
A note to check the titles and content and ensure spacing and capitalization does not include any typos

AAraKKe commented on cloudgen_firewall/assets/dashboards/cloudgen_firewall_overview.json:
```diff
@@ -0,0 +1,6282 @@
+            "id": 8481596997566296,
+            "definition": {
+                "title": "Barracuda CloudGen Firewall",
+                "banner_img": "/api/v2/images/72076454-0aca-4881-a73b-ff421931b5dd",
```
I might be wrong but I do not think this logo is properly added to this dashboard. Where is this url from? I deployed the assets in our staging environment and I cannot see the logo in the dashboard (image attached). The logo is available to be used [here](https://druids.us1.prod.dog/logos/logo-gallery?query=cloudgen&variant=large#cloudgen-firewall). 

**note**: the guidelines state to use a banner with an image but we have agreed to use light and dark variant accordingly so the dashboards adapt to the users theme. The docs will be updated soon to reflect this.

Could you go thorugh the Dashboard and ensure it aligns with our [Dashboard guidelines](https://datadoghq.dev/integrations-core/guidelines/dashboards/#best-practices)? The alignment seems to be off and the size of the different widgets is not matched through the dashboad.

<img width="3143" height="1742" alt="Image" src="https://github.com/user-attachments/assets/cfc005c7-32d3-446b-b3de-58e33ef61a12" />

vinodkumar-sacumen replied:
@AAraKKe We dont have access to the logo link. Would request you please share the access or share the downloadable link. Thanks

AAraKKe replied:
I thought it was open, sorry. These are the paths to be used:
- Light theme: static/images/logos/cloudgen-firewall_large.svg
- Dark theme: static/images/logos/cloudgen-firewall_reversed_large.svg

I think that should be enough and avoid getting outside the app to get the image, if the image does not load, add the full url with the domain `https://static.datadoghq.com`.

vinodkumar-sacumen replied:
Thanks! Added the Light theme and updated few color code and looks based on the guidelines. Please have a look. Thanks

AAraKKe replied:
Thanks!

AAraKKe reviewed (changes requested):
Added a couple of comments. The main one is about the dashboard. Thanks a lot!

AAraKKe commented on cloudgen_firewall/assets/logs/cloudgen_firewall.yaml:
```diff
@@ -0,0 +1,561 @@
+    path: cloudgen_firewall.session_status
+    source: log
+    type: string
+pipeline:
```
I am looking at the pipeline and I don't think this pipeline would work. The reasons are:

- We have remapers at the beginning of the pipeline, before we have defined any of the fields we are going to be using on the different remappers and nested pipelines below.
- Different nested pipelines seem to redefine what each field means. We would need to move the remappers at the end or once remapped the field could be updated and not get the latest value. For example, we first remap component to service, then redefine what component is, service would never be updated.

This pipeline was already present in staging but I saw some syntax errors which makes me think this might be outdated. I am trying to re-deploy it and submit some sample logs. With the one in staging at the moment no logs are processed by the pipeline.

vinodkumar-sacumen replied:
Yes. The log entries being populated work with fields as expected as shown.
<img width="1178" height="631" alt="image" src="https://github.com/user-attachments/assets/dc493e96-a213-4d00-8e6a-3741606b5d5a" />

vinodkumar-sacumen replied:
Plus, FYI @ieguinoa had shared the same file to be updated in the PR as this version was working.
![Uploading image.png…]()

vinodkumar-sacumen commented:
> Sorry for the wait @vinodkumar-sacumen, I added some comments below about the structure of the pipeline and nested pipeline. At the end what we want to achieve is something like this
> 
> <img alt="Image" width="1003" height="1034" src="https://private-user-images.githubusercontent.com/2953658/487252029-3cd9258c-4ec9-4e3f-9ab1-32a0a33425c8.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTc1MDEyMTQsIm5iZiI6MTc1NzUwMDkxNCwicGF0aCI6Ii8yOTUzNjU4LzQ4NzI1MjAyOS0zY2Q5MjU4Yy00ZWM5LTRlM2YtOWFiMS0zMmEwYTMzNDI1YzgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDkxMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA5MTBUMTA0MTU0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDFhMzYzYWI4MTMxZTI3ZmY5ZmZlN2RlMWI3ZWNkNmNmMTAxMDA0MzE3ZGI1OTRjMzU5ZDc0ZTgwZTg3Y2MyYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.IuhE5CFsypTRtzCa6xFQj2ay_AJ-2kq5wPePZZ7TwjQ">
> Let me know if you have any question.
> 
> At the moment, the yaml file in this PR and the screenshot you shared above do not match. You can see how the remappers are first and this is also what we see when we deploy this pipeline to our staging environment. See the image below. Not sure if this PR is out of sync with yours.
> 
> <img alt="Image" width="2000" height="680" src="https://private-user-images.githubusercontent.com/2953658/487252637-d8df2bac-3576-413f-bde3-a27dc1e41963.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTc1MDEyMTQsIm5iZiI6MTc1NzUwMDkxNCwicGF0aCI6Ii8yOTUzNjU4LzQ4NzI1MjYzNy1kOGRmMmJhYy0zNTc2LTQxM2YtYmRlMy1hMjdkYzFlNDE5NjMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDkxMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA5MTBUMTA0MTU0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTk5MmU1Y2MwMDcxYzNhZjBjYTU3N2ZiZjhkOTU2MjNmZjI3ZmRmZDNlN2Y5MmE0NThhOWQyM2QyYzA1ODZmZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.QKZZY169YPQKeexgVFPyf-KnyrClOCHExpQe9k9oDgM">

Basically, we need to change the order.

AAraKKe commented on cloudgen_firewall/assets/logs/cloudgen_firewall.yaml:
```diff
@@ -0,0 +1,576 @@
+metric_id: cloudgen-firewall
+backend_only: false
+installation_sources:
+  - barracuda_cloudgen_logs
```
This needs to match the example in the `conf.yaml.example` provided.  If a customer uses the integration as it is the pipeline would not work because the configuration provided as an exmaple sets a different source that the one in the pipeline.

I would say that what makes the most sense is to have the pipeline react to a source being `cloudget_firewall` as this is the closes to the integration ID. But having the source in the example config set to `barracuda_cloudget_logs` works as well.

vinodkumar-sacumen replied:
updated.

AAraKKe commented on cloudgen_firewall/assets/logs/cloudgen_firewall_tests.yaml:
Please, make sure to include as much testa as possible. We have simplified this a lot to be able to have the pipeline deployed to staging but we need to bring back all tests cases that were here before and ensure they are parsed as we expect them to be parsed.

AAraKKe reviewed (changes requested):
Thanks for the update on the pipeline, we are nearly there! 

I added some comments that should fix the issues we have discussed offline.

vinodkumar-sacumen commented on cloudgen_firewall/assets/logs/cloudgen_firewall.yaml:
```diff
@@ -0,0 +1,576 @@
+        - type: grok-parser
+          name: Box Auth Access
+          enabled: true
+          source: nested_message
```
Tested using @nested_message, it's not working. Hence, using just nested_message as is.

vinodkumar-sacumen commented on cloudgen_firewall/assets/logs/cloudgen_firewall.yaml:
```diff
@@ -0,0 +1,576 @@
+      targetType: attribute
+      preserveSource: true
+      overrideOnConflict: false
+    - type: attribute-remapper
```
Yes I have removed it. It makes more sense for me as well.

AAraKKe commented on cloudgen_firewall/assets/logs/cloudgen_firewall.yaml:
```diff
@@ -0,0 +1,511 @@
+    groups:
+      - Barracuda CloudGen Firewall
+    name: component
+    path: component
```
@vinodkumar-sacumen I think this comment was never addressed, could you please make sure that the facets have the proper naming?

AAraKKe commented on cloudgen_firewall/datadog_checks/cloudgen_firewall/data/conf.yaml.example:
```diff
@@ -0,0 +1,19 @@
+# logs:
+#   - type: file
+#     path: /var/log/cloudgen_firewall.log
+#     source: barracuda_cloudgen_logs
```
This needs to be udpated in the spec, not directly in the example file. Once the spec is updated, run `ddev validate config cloudgen_firewall -s`

vinodkumar-sacumen commented on cloudgen_firewall/assets/logs/cloudgen_firewall.yaml:
```diff
@@ -0,0 +1,457 @@
+            supportRules: ""
+            matchRules: >-
+              autoFilledRule1
+              %{word:source}\s+%{word:service}?:\s+%{word:sub_type}?:\s+%{greedyData:message}
```
Can you please cross check. I dont see any service in the parsed result. plus we had already removed component to service remapper as well. 
<img width="1227" height="553" alt="image" src="https://github.com/user-attachments/assets/7f5811b6-9b93-4f9e-a560-852cbcb1443b" />

AAraKKe commented on cloudgen_firewall/assets/logs/cloudgen_firewall.yaml:
```diff
@@ -0,0 +1,457 @@
+    facetType: list
+    groups:
+      - Barracuda CloudGen Firewall
+    name: component
```
Keep capitalization consistent in all namens in the facets, as well as plural form.

```
component -> Component
IPs -> IP
process -> Process
session_status -> Session Status
```

Make sure to review this in all facet names.

AAraKKe reviewed (changes requested):
Just adde dan extra comment now that I saw you updated the facet paths.

buraizu commented on cloudgen_firewall/README.md:
```diff
@@ -0,0 +1,78 @@
+4. Check the Datadog [Live Tail][7] in Datadog for logs from the source and service you defined in the `conf.yaml` file.
+5. After following these steps, you can create a test log on the firewall by triggering an event.
+6. Check for tags or facets to use them for better filtering based on the required data.
+
+### Metrics
```
Suggesting a reordering of headings here for logical flow

buraizu commented on cloudgen_firewall/README.md:
```diff
@@ -0,0 +1,78 @@
+Need help? Contact [Datadog support][3].
+
+[1]: https://www.barracuda.com/products/network-protection/cloudgen-firewall
+[2]: https://app.datadoghq.com/account/settings/agent/latest
+[3]: https://docs.datadoghq.com/help/
+[4]: https://docs.datadoghq.com/agent/guide/agent-commands/#start-stop-and-restart-the-agent
+[5]: https://app.datadoghq.com/integrations?search=barracuda_cloudgen_firewall
+[6]: https://github.com/DataDog/integrations-core/blob/master/cloudgen_firewall/assets/service_checks.json
+[7]: https://app.datadoghq.com/logs/livetail
```
```suggestion
[2]: /account/settings/agent/latest
[3]: https://docs.datadoghq.com/help/
[4]: https://docs.datadoghq.com/agent/guide/agent-commands/#start-stop-and-restart-the-agent
[5]: /integrations?search=barracuda_cloudgen_firewall
[6]: https://github.com/DataDog/integrations-core/blob/master/cloudgen_firewall/assets/service_checks.json
[7]: /logs/livetail
```

buraizu replied:
Links that lead to the Datadog app need to be relative, so that they work regardless of the Datadog site being used.

Additionally, links `[5]` and `[6]` are not being used in the page. Could you confirm if this is expected, and remove them if so?

buraizu commented on cloudgen_firewall/manifest.json:
```diff
@@ -0,0 +1,63 @@
+    "configuration": "README.md#Setup",
+    "support": "README.md#Support",
+    "changelog": "CHANGELOG.md",
+    "description": "Barracuda CloudGen Firewall is a NGFW that protect networks and Internet traffic",
```
```suggestion
    "description": "Barracuda CloudGen Firewall is an NGFW that protects networks and Internet traffic",
```

buraizu commented on cloudgen_firewall/manifest.json:
```diff
@@ -0,0 +1,63 @@
+      },
+      {
+        "media_type": "image",
+        "caption": "Some more Events types of CloudGen Firewall",
```
```suggestion
        "caption": "Some more Event types of CloudGen Firewall",
```

buraizu commented on cloudgen_firewall/README.md:
```diff
@@ -0,0 +1,78 @@
+
+### Events
+
+The cloudgen_firewall integration includes log events such as failed logins and rule hits.
```
```suggestion
The cloudgen_firewall integration does not include any events.
```

buraizu replied:
Changing this because `creates_events` is `false` in the manifest file. This section is for events going to Datadog's [Event Management](https://docs.datadoghq.com/service_management/events/) platform, rather than Log Management.

buraizu commented on cloudgen_firewall/README.md:
```diff
@@ -0,0 +1,78 @@
+    `sudo netstat -tunlp | grep 514`
+    If using TCP and UDP listeners, use the following command:
+    `sudo lsof -i :514`
+2. Confirm logs are reaching the Agent from the correct log source.
```
```suggestion
2. Confirm logs are reaching the Agent from the correct log source:
```

buraizu reviewed (changes requested):
Requesting some additional edits on the updated README file

vinodkumar-sacumen commented on cloudgen_firewall/images/cloudgen_firewall_overview.png:
We have added the images based on the ratio mentioned in the checks. If we change it, the checks will start to fail. @shrey4b

AAraKKe commented on cloudgen_firewall/assets/logs/cloudgen_firewall.yaml:
```diff
@@ -0,0 +1,457 @@
+      grok:
+        supportRules: ""
+        matchRules: |
+          rule1 %{hostname}/box_%{data:component}(_%{data})?
```
I see this has not been updated yet in the PR but got a message saying everything was updated. Can we have the PR in sync with the latest version?

vinodkumar-sacumen replied:
You are right, The PR doesn't seem to be updated. I am force pushing the changes. For some reason it is not giving upto date.

AAraKKe commented on cloudgen_firewall/assets/logs/cloudgen_firewall.yaml:
```diff
@@ -0,0 +1,457 @@
+    groups:
+      - Barracuda CloudGen Firewall
+    name: Process
+    path: cloudgen_firewall.process
```
The path in facets represents the field they facets are going to filter on. I just realized that none of the fields defined in the logs pipeline is prepended by the `cloudgen_firewall` prefix. Every field we defined as a custom field should be prepended by the prefix `cloudgen_firewall` to show that those fields are specific to this integration and for facets to work.

We can have a remapper at the end of the pipeline redefining the fields we have with the proper prefix and keep simpler names during the pipeline process.

vinodkumar-sacumen replied:
Yes. We have to prepened the factes path with cloudgen_firewall to have unique facets. 
We are unable to update the facet path in the account pipeline. We update them in the pipeline yaml file manually. Let me know if you see any path is missing to have  a prefix. Thanks!

AAraKKe reviewed (changes requested):
I was taking a look at the pipeline in staging and noticed that the changes discussed were not yet applied. I realized the PR was not updated with the recent changes discussed. Can we address the comments and update the PR?

AAraKKe reviewed (dismissed):
Thanks for the updates @vinodkumar-sacumen !!
